File: templates/problems/07_trees/08_level_order/go/level_order_test.go
package level_order

import (
	"reflect"
	"testing"
)

func TestLevelOrder(t *testing.T) {
	tests := []struct {
		name string
		vals []int
		want [][]int
	}{
		{"example 1", []int{3, 9, 20, -101, -101, 15, 7}, [][]int{{3}, {9, 20}, {15, 7}}},
		{"example 2", []int{1}, [][]int{{1}}},
		{"empty tree", []int{}, nil},
		{"two levels", []int{1, 2, 3}, [][]int{{1}, {2, 3}}},
		{"left only", []int{1, 2, -101, 3}, [][]int{{1}, {2}, {3}}},
		{"full tree", []int{1, 2, 3, 4, 5, 6, 7}, [][]int{{1}, {2, 3}, {4, 5, 6, 7}}},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			root := buildTreeHelper(tt.vals)
			got := levelOrder(root)
			if !reflect.DeepEqual(got, tt.want) {
				t.Errorf("levelOrder() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: templates/problems/07_trees/08_level_order/typescript/level_order.test.ts
import { describe, it, expect } from "vitest";
import { buildTree } from "./helpers";
import { levelOrder } from "./level_order";

describe("levelOrder", () => {
    it.each([
        { name: "example 1", vals: [3, 9, 20, -101, -101, 15, 7], want: [[3], [9, 20], [15, 7]] },
        { name: "example 2", vals: [1], want: [[1]] },
        { name: "empty tree", vals: [], want: [] },
        { name: "two levels", vals: [1, 2, 3], want: [[1], [2, 3]] },
        { name: "left only", vals: [1, 2, -101, 3], want: [[1], [2], [3]] },
        { name: "full tree", vals: [1, 2, 3, 4, 5, 6, 7], want: [[1], [2, 3], [4, 5, 6, 7]] },
    ])("case: $name", ({ vals, want }) => {
        const root = buildTree(vals);
        expect(levelOrder(root)).toEqual(want);
    });
});
